workloadccl: split slow TestAllRegisteredSetup

hueypark opened the pull request:
This splits TestAllRegisteredSetup for each setup. It allows test runtime faster not only in a single machine, can distribute execute when running on CI.

And new `testgen` cmd makes this productivity improvement job easy.

Release note: None

Related to #37028

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/39107)
<!-- Reviewable:end -->

RaduBerinde commented:
Hi Huey,
Thank you for contributing. Moving go code in strings and generating code should be a last resort though. If all we want is parallelization of the tests, we can achieve that with subtests and `RunParallel` (see this patch for example: https://gist.github.com/RaduBerinde/051e83d5d8863b6e3bc91379356e74ec). But as Andy pointed out in https://github.com/cockroachdb/cockroach/issues/37028#issuecomment-495065529, just parallelizing tests won't help too much with making CI or `make test` faster because these already run tests for multiple packages in parallel. It does help when you run `make test PKG=..` on a single package though.

hueypark commented:
Thank you for your review. I will be back soon with the version of Andy's comments applied.

hueypark committed:
workloadccl: split slow TestAllRegisteredSetup

This splits TestAllRegisteredSetup for each setup. It allows test runtime faster not only in a single machine, can distribute execute when running on CI.

And new `testgen` cmd makes this productivity improvement job easy.

Release note: None
